### ISSUE TITLE
Is it possible when using openTelemetry with automatic agent and manual setting extensions up simutanously?

### ISSUE THREAD

**Issue by Master-Sword-Ext** [open-telemetry/opentelemetry-java-instrumentation#9567]
### Is your feature request related to a problem? Please describe.

Currently, I just can't use the extension machinism rightly for the existence of AgentExtension Classloader. The documentation give the reason for security which extension agent shouldn't intervene the main application runtime behavior. However, for real senario, you can't just modified the id generator, span processor without the application logic. On the other way, In manual instrumentation example, I can wire up the Tracer using SdkTracerBuilderProvider, with my own id generator, span processor etc. In manual way, this behavior is just like my applicaiton code. But manual instrumentation need a lot of effort to deal with third party library which automatic instrumentation offers naturaly.  So is there any solution to this problem?

### Describe the solution you'd like

In my opinion, to solve the problem, the custom config whatever in automatic or manual instrumentation should be the same. Another way to solve the problem is modifying the agent extension classloader, make it lightweight or swichable, let the user to choose more strict or less strict.
To summarize
1. Make automatic and manual instrumentation behavior the same, which chould be fullfill by initialize the GlobalTracer second chance in the application code.
2. Make agent extension classloader optional, let user choose how to customize the extension part

Lastly, I use the OpenTelemetry for years, the way it describes the distributed system is the best as I know, hope it moves better

**Comment by breedx-splk** [comment]
Hey @Master-Sword-Ext. I'm having a hard time following the problem that you're having. It sounds like you're having trouble using an agent extension that you've written? You mentioned an id generator, is this for trace IDs / span IDs or something else? 

Anyway, you should be able to leverage autoconfigure / `@AutoService` to include an instance of `AutoConfigurationCustomizerProvider` in your agent extension. That `AutoConfigurationCustomizerProvider` will receive an instance of `SdkTracerProviderBuilder` in its configure() method, and you can set the id generator on that.

**Comment by Master-Sword-Ext** [comment]
@breedx-splk thank you for your kind reply. I had read the offical doc of the extension usage, the [DemoAutoConfiger](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/examples/extension/src/main/java/com/example/javaagent/DemoAutoConfigurationCustomizerProvider.java) here.  
First of all,  i had tried the method in offical doc, it forces use agentClassloader to load my custome id generator, which however should also be load by my bussiness classloader, thus causes the conflict. This kind of method can't meet my needs. 
Secondly, If I use the custom instrumentation of otel, I can use my custome id generator with my custom classloader very well, but this method make me lost the automatic part of the instrumentation. 

So, for the sake of security, agent classloader current mechanism is reasonable, so is there anyway to solve my problem above?

**Comment by breedx-splk** [comment]
Whoops, I had pasted the wrong link above. Fixed. 

Anyway, so your problem is that the id generator is on the agent classpath and you also need it in your application classpath?

**Comment by Master-Sword-Ext** [comment]
@breedx-splk sorry for replying so late. Yes, you can understand it this way, the id generator need use the application specific info. 😄 😄

**Comment by xiepuhuan** [comment]
> @breedx-splk sorry for replying so late. Yes, you can understand it this way, the id generator need use the application specific info. 😄 😄

With reflection, you can use the system classloader in the Id generator to obtain the application information
